stream-chat-android-client/src/main/java/io/getstream/chat/android/client/audio/NativeMediaPlayer.kt

```diff
@@ -239,45 +293,98 @@ internal class NativeMediaPlayerImpl(
         SecurityException::class,
         IllegalStateException::class,
     )
-    override fun setDataSource(path: String) = mediaPlayer.setDataSource(path)
+    override fun setDataSource(path: String) {
+        if (DEBUG) logger.d { "[setDataSource] path: $path" }
+        mediaPlayer.setDataSource(path)
+        state = NativeMediaPlayerState.INITIALIZED
+    }
 
     @Throws(IllegalStateException::class)
-    override fun prepareAsync() = mediaPlayer.prepareAsync()
+    override fun prepareAsync() {
+        if (DEBUG) logger.d { "[prepareAsync] no args" }
+        mediaPlayer.prepareAsync()
+        state = NativeMediaPlayerState.PREPARING
+    }
 
     @Throws(IOException::class, IllegalStateException::class)
-    override fun prepare() = mediaPlayer.prepare()
+    override fun prepare() {
+        if (DEBUG) logger.d { "[prepare] no args" }
+        mediaPlayer.prepare()
+        state = NativeMediaPlayerState.PREPARED
+    }
 
     @Throws(IllegalStateException::class)
-    override fun seekTo(msec: Int) = mediaPlayer.seekTo(msec)
+    override fun seekTo(msec: Int) {
+        if (DEBUG) logger.d { "[seekTo] msec: $msec" }
+        mediaPlayer.seekTo(msec)
+    }
 
     @Throws(IllegalStateException::class)
-    override fun start() = mediaPlayer.start()
+    override fun start() {
+        if (DEBUG) logger.d { "[start] no args" }
+        mediaPlayer.start()
+        state = NativeMediaPlayerState.STARTED
+    }
 
     @Throws(IllegalStateException::class)
-    override fun pause() = mediaPlayer.pause()
+    override fun pause() {
+        if (DEBUG) logger.d { "[pause] no args" }
+        mediaPlayer.pause()
+        state = NativeMediaPlayerState.PAUSED
+    }
 
     @Throws(IllegalStateException::class)
-    override fun stop() = mediaPlayer.stop()
+    override fun stop() {
+        if (DEBUG) logger.d { "[stop] no args" }
+        mediaPlayer.stop()
+        state = NativeMediaPlayerState.STOPPED
+    }
 
-    override fun reset() = mediaPlayer.reset()
+    override fun reset() {
+        if (DEBUG) logger.d { "[reset] no args" }
+        mediaPlayer.reset()
```

Comment:
What do you think about wrapping this into defensive `try-catch` block and setting `state = NativeMediaPlayerState.ERROR` in case it fails.

Maybe also wrap `if (DEBUG) { // Log inside... }` (old habit) :)

Same for the other functions.

Reply:
Actually, I'd love to have it crashing, cause those crashes are due to the nature of `MediaPlayer` having a state machine in a wrong state. So for customers it would look like nothing crashes, but nothing works properly and for us this would be harder to debug 🙂 

`if (DEBUG) { // Log inside... }` I intentionally made it a one-liner to have it compact, cause `spotless` will transform it to
```kotlin
if (DEBUG) { 
    // Log inside... 
}
```